models/classes/resources/ListResourceLookup.php

```diff
@@ -74,7 +79,13 @@ private function searchByString($searchString, $rootClass, $offset, $limit)
         /** @var ResultSet $result */
         $result = $searchService->query($searchString, $rootClass, $offset, $limit);
         $count = $result->getTotalCount();
-        return $this->format($result, $count, $offset, $limit);
+
+        $ids = [];
+        foreach ($result as $item) {
+            $ids[] = $item['id'];
```

Comment:
If you decide not to go deeper and address the root cause (see https://github.com/oat-sa/tao-core/pull/3702#pullrequestreview-1220759737) for one reason or another, let's at least make this piece of code compatible with all search-service implementations.
```suggestion
            $ids[] = $item['id'] ?? $item;
```

Reply:
In a regular tao installation, using the search bar I could not reach this lines, so I assume only specific cases like INVALSI are using it. 

Changing these lines did not break the advanced search for me